### ISSUE TITLE
randr changes cause startup failure

### ISSUE THREAD

**Issue by tgbugs** [stumpwm/stumpwm#615]
I have encountered an issue that appears to be caused by https://github.com/stumpwm/stumpwm/pull/570 (git bisect below). The error itself is quite cryptic because it seems that the error handling in clx doesn't know what to do with the crtc error code that it gets. The lines that are the acute cause of the issue are [here](https://github.com/sharplispers/clx/blob/4b9c2026c98e81af5734106dc19a6609c2cbe5ab/input.lisp#L1482-L1483). `error-decode-function` is nil because clx doesn't know what to do with error-key `XLIB::CRTC`. If you route around that error by checking if `error-decode-function` is nil then you are off into uncharted territory. I will do a bit more testing and report back on the issue. There are a number of weird things in my setup that could potentially be causing this (screen rotated 90 degrees, twinview, two connectors pugged into the same monitor, etc.).

edit1: Confirming that reverting 84a541c9cc2081dafef590f0c0411f7e3a5e748e solves the issue.

```
git bisect start
# bad: [3b442c4c3bb081ba115c4d8aa939c8d80435aaa6] Don't update the modeline on mouse scroll events
git bisect bad 3b442c4c3bb081ba115c4d8aa939c8d80435aaa6
# good: [1b89469bd9f6e7672788d42ac8269448b8cde159] Refactor the logic behind *FLOAT-WINDOW-MODIFIER*
git bisect good 1b89469bd9f6e7672788d42ac8269448b8cde159
# good: [d39fed88674080043957a87b5d987a27b63ac260] Merge pull request #540 from stumpwm/simplify-all-modifiers
git bisect good d39fed88674080043957a87b5d987a27b63ac260
# good: [4d4b6898ab7e4caefc11cc8f8b30007c63d7111a] Merge pull request #544 from epanji/float-window-border
git bisect good 4d4b6898ab7e4caefc11cc8f8b30007c63d7111a
# bad: [353f903eb8598eb61dda854272e82948a73ea9e8] Export the documented *new-head-hook*
git bisect bad 353f903eb8598eb61dda854272e82948a73ea9e8
# bad: [da319cc17d48c9745a40ca6e75a0c2d28b06aedc] Merge pull request #570 from ralt/pr/xrandr
git bisect bad da319cc17d48c9745a40ca6e75a0c2d28b06aedc
# bad: [84a541c9cc2081dafef590f0c0411f7e3a5e748e] Generate the heads using Xrandr instead of Xinerama.
git bisect bad 84a541c9cc2081dafef590f0c0411f7e3a5e748e
```

```
 xinit ~/.xinitrc-stumpwm -- :2.0

X.Org X Server 1.20.4
X Protocol Version 11, Revision 0
Build Operating System: Linux 4.19.5-gentoo x86_64 Gentoo
Current Operating System: Linux athena 4.19.5-gentoo #1 SMP Sat Dec 1 13:29:27 PST 2018 x86_64
Kernel command line: rootfstype=ext4 nomodeset
Build Date: 05 May 2019  07:16:36PM
 
Current version of pixman: 0.38.4
	Before reporting problems, check http://wiki.x.org
	to make sure that you have the latest version.
Markers: (--) probed, (**) from config file, (==) default setting,
	(++) from command line, (!!) notice, (II) informational,
	(WW) warning, (EE) error, (NI) not implemented, (??) unknown.
(==) Log file: "/var/log/Xorg.2.0.log", Time: Sun May 26 18:20:51 2019
(==) Using config file: "/etc/X11/xorg.conf"
(==) Using config directory: "/etc/X11/xorg.conf.d"
(==) Using system config directory "/usr/share/X11/xorg.conf.d"
The XKEYBOARD keymap compiler (xkbcomp) reports:
> Warning:          Unsupported high keycode 372 for name <I372> ignored
>                   X11 cannot support keycodes above 255.
>                   This warning only shows for the first high keycode.
> Internal error:   Could not resolve keysym XF86MonBrightnessCycle
Errors from xkbcomp are not fatal to the X server

debugger invoked on a UNDEFINED-FUNCTION in thread
#<THREAD "main thread" RUNNING {10005705B3}>:
  The function COMMON-LISP:NIL is undefined.

Type HELP for debugger help, or (SB-EXT:EXIT) to exit from SBCL.

restarts (invokable by number or by possibly-abbreviated name):
  0: [CONTINUE      ] Retry calling NIL.
  1: [USE-VALUE     ] Call specified function.
  2: [RETURN-VALUE  ] Return specified values.
  3: [RETURN-NOTHING] Return zero values.
  4: [QUIT-STUMPWM  ] Quit Stumpwm
  5: [ABORT         ] Exit from the current thread.

("undefined function" #<XLIB:DISPLAY :2 (The X.Org Foundation R12004000)> #S(XLIB::REPLY-BUFFER :SIZE 32 :IBUF8 #(0 146 67 0 0 0 0 0 20 0 139 0 ...) :NEXT NIL :DATA-SIZE 32))
0]
```
```
Backtrace:
  0: ("undefined function" #<XLIB:DISPLAY :2 (The X.Org Foundation R12004000)> #S(XLIB::REPLY-BUFFER :SIZE 32 :IBUF8 #(0 146 64 0 0 0 ...) :NEXT NIL :DATA-SIZE 32))
  1: (XLIB::MAKE-ERROR #<XLIB:DISPLAY :2 (The X.Org Foundation R12004000)> #S(XLIB::REPLY-BUFFER :SIZE 32 :IBUF8 #(0 146 64 0 0 0 ...) :NEXT NIL :DATA-SIZE 32) NIL)
  2: (XLIB::READ-REPLY #<XLIB:DISPLAY :2 (The X.Org Foundation R12004000)> #S(XLIB::PENDING-COMMAND :SEQUENCE 64 :REPLY-BUFFER NIL :PROCESS #<SB-THREAD:THREAD "repl-thread" RUNNING {10063C0413}> :NEXT NIL)..
  3: (XLIB:RR-GET-CRTC-INFO #<XLIB:DISPLAY :2 (The X.Org Foundation R12004000)> 0 3767909932 LIST)
  4: (STUMPWM::OUTPUT->HEAD 667 5)
  5: (STUMPWM::MAKE-SCREEN-RANDR-HEADS #<XLIB:WINDOW :2 2A7>)
  6: (STUMPWM::MAKE-SCREEN-HEADS #<unused argument> #<XLIB:WINDOW :2 2A7>)
  7: (STUMPWM::INIT-SCREEN #<XLIB:SCREEN :2.0 3120x1920x24 TRUE-COLOR> 0 "")
  8: (STUMPWM::STUMPWM-INTERNAL ":2.0")
  9: (STUMPWM:STUMPWM ":2.0")
 10: (SB-INT:SIMPLE-EVAL-IN-LEXENV (STUMPWM:STUMPWM ":2.0") #<NULL-LEXENV>)
 11: (EVAL (STUMPWM:STUMPWM ":2.0"))
 12: (SWANK::EVAL-REGION "(stumpwm:stumpwm \":2.0\") ..)
 13: ((LAMBDA NIL :IN SWANK-REPL::REPL-EVAL))
 14: (SWANK-REPL::TRACK-PACKAGE #<CLOSURE (LAMBDA NIL :IN SWANK-REPL::REPL-EVAL) {10063CA07B}>)
 15: (SWANK::CALL-WITH-RETRY-RESTART "Retry SLIME REPL evaluation request." #<CLOSURE (LAMBDA NIL :IN SWANK-REPL::REPL-EVAL) {10063CA01B}>)
 16: (SWANK::CALL-WITH-BUFFER-SYNTAX NIL #<CLOSURE (LAMBDA NIL :IN SWANK-REPL::REPL-EVAL) {10063C9FFB}>)
 17: (SWANK-REPL::REPL-EVAL "(stumpwm:stumpwm \":2.0\") ..)
 18: (SB-INT:SIMPLE-EVAL-IN-LEXENV (SWANK-REPL:LISTENER-EVAL "(stumpwm:stumpwm \":2.0\") ..)
 19: (EVAL (SWANK-REPL:LISTENER-EVAL "(stumpwm:stumpwm \":2.0\") ..)
 20: (SWANK:EVAL-FOR-EMACS (SWANK-REPL:LISTENER-EVAL "(stumpwm:stumpwm \":2.0\") ..)
 21: (SWANK::PROCESS-REQUESTS NIL)
 22: ((LAMBDA NIL :IN SWANK::HANDLE-REQUESTS))
 23: ((LAMBDA NIL :IN SWANK::HANDLE-REQUESTS))
 24: (SWANK/SBCL::CALL-WITH-BREAK-HOOK #<FUNCTION SWANK:SWANK-DEBUGGER-HOOK> #<CLOSURE (LAMBDA NIL :IN SWANK::HANDLE-REQUESTS) {10063C818B}>)
 25: ((FLET SWANK/BACKEND:CALL-WITH-DEBUGGER-HOOK :IN "quicklisp/dists/quicklisp/software/slime-v2.23/swank/sbcl.lisp") #<FUNCTION SWANK:SWANK-DEBUGGER-HOOK> #<CLOSURE (LAMBDA NIL :I..
 26: (SWANK::CALL-WITH-BINDINGS ((*STANDARD-INPUT* . #1=#<SWANK/GRAY::SLIME-INPUT-STREAM {10061BAC23}>) (*STANDARD-OUTPUT* . #2=#<SWANK/GRAY::SLIME-OUTPUT-STREAM {10062B08E3}>) (*TRACE-OUTPUT* . #2#) (*ERR..
 27: (SWANK::HANDLE-REQUESTS #<SWANK::MULTITHREADED-CONNECTION {10058AEFD3}> NIL)
 28: ((FLET SB-UNIX::BODY :IN SB-THREAD::INITIAL-THREAD-FUNCTION-TRAMPOLINE))
 29: ((FLET "WITHOUT-INTERRUPTS-BODY-4" :IN SB-THREAD::INITIAL-THREAD-FUNCTION-TRAMPOLINE))
 30: ((FLET SB-THREAD::WITH-MUTEX-THUNK :IN SB-THREAD::INITIAL-THREAD-FUNCTION-TRAMPOLINE))
 31: ((FLET "WITHOUT-INTERRUPTS-BODY-1" :IN SB-THREAD::CALL-WITH-MUTEX))
 32: (SB-THREAD::CALL-WITH-MUTEX #<CLOSURE (FLET SB-THREAD::WITH-MUTEX-THUNK :IN SB-THREAD::INITIAL-THREAD-FUNCTION-TRAMPOLINE) {7FAE623BED7B}> #<SB-THREAD:MUTEX "thread result lock" owner: #<SB-THREAD:THR..
 33: (SB-THREAD::INITIAL-THREAD-FUNCTION-TRAMPOLINE #<SB-THREAD:THREAD "repl-thread" RUNNING {10063C0413}> NIL #<CLOSURE (LAMBDA NIL :IN SWANK-REPL::SPAWN-REPL-THREAD) {10063C03BB}> NIL)
 34: ("foreign function: call_into_lisp")
 35: ("foreign function: new_thread_trampoline")
```

```
xdpyinfo 
name of display:    :0.0
version number:    11.0
vendor string:    The X.Org Foundation
vendor release number:    12004000
X.Org version: 1.20.4
maximum request size:  16777212 bytes
motion buffer size:  256
bitmap unit, bit order, padding:    32, LSBFirst, 32
image byte order:    LSBFirst
number of supported pixmap formats:    7
supported pixmap formats:
    depth 1, bits_per_pixel 1, scanline_pad 32
    depth 4, bits_per_pixel 8, scanline_pad 32
    depth 8, bits_per_pixel 8, scanline_pad 32
    depth 15, bits_per_pixel 16, scanline_pad 32
    depth 16, bits_per_pixel 16, scanline_pad 32
    depth 24, bits_per_pixel 32, scanline_pad 32
    depth 32, bits_per_pixel 32, scanline_pad 32
keycode range:    minimum 8, maximum 255
focus:  window 0x4e00006, revert to PointerRoot
number of extensions:    29
    BIG-REQUESTS
    Composite
    DAMAGE
    DOUBLE-BUFFER
    DPMS
    DRI2
    GLX
    Generic Event Extension
    MIT-SCREEN-SAVER
    MIT-SHM
    NV-CONTROL
    NV-GLX
    Present
    RANDR
    RECORD
    RENDER
    SHAPE
    SYNC
    X-Resource
    XC-MISC
    XFIXES
    XFree86-DGA
    XFree86-VidModeExtension
    XINERAMA
    XINERAMA
    XInputExtension
    XKEYBOARD
    XTEST
    XVideo
default screen number:    0
number of screens:    1
```
```
Screen 0: minimum 8 x 8, current 3120 x 1920, maximum 16384 x 16384
DVI-I-0 disconnected (normal left inverted right x axis y axis)
DVI-I-1 connected primary 1200x1920+0+0 left (normal left inverted right x axis y axis) 519mm x 324mm
   1920x1200     59.95*+
   1680x1050     59.95  
   1600x1200     60.00  
   1280x1024     75.02    60.02  
   1152x864      75.00  
   1024x768      75.03    60.00  
   800x600       75.00    60.32  
   640x480       75.00    59.94  
HDMI-0 disconnected (normal left inverted right x axis y axis)
DP-0 disconnected (normal left inverted right x axis y axis)
DVI-D-0 connected 1920x1080+1200+300 (normal left inverted right x axis y axis) 531mm x 299mm
   1920x1080     60.00 + 144.00*  119.98    99.93    84.90  
   1680x1050     59.95  
   1440x900      59.89  
   1280x1024     75.02    60.02  
   1280x960      60.00  
   1280x800      59.81  
   1280x720      60.00  
   1152x864      75.00  
   1024x768      75.03    70.07    60.00  
   800x600       75.00    72.19    60.32    56.25  
   640x480       75.00    59.94  
DP-1 connected (normal left inverted right x axis y axis)
   1920x1080     60.00 + 144.00   119.98    99.93    84.90    59.94    50.00  
   1680x1050     59.95  
   1440x900      59.89  
   1280x1024     75.02    60.02  
   1280x960      60.00  
   1280x800      59.81  
   1280x720      60.00    59.94    50.00  
   1152x864      75.00  
   1024x768      75.03    70.07    60.00  
   800x600       75.00    72.19    60.32    56.25  
   720x576       50.00  
   720x480       59.94  
   640x480       75.00    59.94    59.93  
```

**Comment by dbjergaard** [comment]
Thanks for the awesome bug report. I should be able to fix this next time I have some free cycles.

**Comment by scottjad** [comment]
I think this is the same issue as #586 and has been fixed by #611.

**Comment by tgbugs** [comment]
@scottjad this is fixed for me now, so closing. Thanks!
